pkg/proc/stack.go

```diff
@@ -438,13 +447,24 @@ func (it *stackIterator) advanceRegs() (callFrameRegs op.DwarfRegisters, ret uin
 	// implicit.
 	// See also the comment in dwarf2_frame_default_init in
 	// $GDB_SOURCE/dwarf2-frame.c
-	if _, ok := it.bi.Arch.(*ARM64); ok {
+	_, isARM64 := it.bi.Arch.(*ARM64)
+	if isARM64 {
```

Comment:
Do `callFrameRegs.AddReg(callFrameRegs.SPRegNum, cfareg)` here instead.

Reply:
The new code variable "isARM64" was only added on the assumption that it is probably cheaper to refer to this variable later rather than re-execute "it.bi.Arch.(*ARM64)" later.

Reply:
`callFrameRegs.AddReg(callFrameRegs.SPRegNum, cfareg)` is better here, because it has same abstract logic in diffrent arch.  
You can see this [pr](https://github.com/go-delve/delve/pull/1884/files#diff-2f765e651083ae7ea60fe0b2437bd7b4L441-R442) about supporting 386 linux.

Reply:
The isARM64 variable is fine, it's the `if` I have an issue with (yes, it's not your fault it exists but while we are here we might as well fix it).

Reply:
OK, will fix it.  I'm busy for the next few days, but should be able to issue a corrected pull request soon.